src/helpers/spotify_manager.ts

```diff
@@ -144,104 +137,111 @@ export default class SpotifyManager {
-                await fs.promises.mkdir(SPOTIFY_PLAYLIST_UNMATCHED_SONGS_DIR);
-            }
+            logger.info(
+                `Finished parsing playlist: ${playlistID} after ${
+                    Date.now() - songMatchStartTime
+                }ms.`
+            );
 
-            const playlistUnmatchedSongsPath = path.resolve(
+            const SPOTIFY_PLAYLIST_UNMATCHED_SONGS_DIR = path.join(
                 __dirname,
-                SPOTIFY_PLAYLIST_UNMATCHED_SONGS_DIR,
-                `${playlistID}-${standardDateFormat(new Date())}.txt`
+                "../../data/spotify_unmatched_playlists"
             );
 
-            await fs.promises.writeFile(
-                playlistUnmatchedSongsPath,
-                unmatchedSongs.join("\n")
-            );
-        }
+            if (unmatchedSongs.length) {
+                if (!(await pathExists(SPOTIFY_PLAYLIST_UNMATCHED_SONGS_DIR))) {
+                    await fs.promises.mkdir(
+                        SPOTIFY_PLAYLIST_UNMATCHED_SONGS_DIR
+                    );
+                }
 
-        if (KmqConfiguration.Instance.persistMatchedSpotifySongs()) {
-            await fs.promises.writeFile(
-                path.resolve(
+                const playlistUnmatchedSongsPath = path.resolve(
+                    __dirname,
                     SPOTIFY_PLAYLIST_UNMATCHED_SONGS_DIR,
-                    `${playlistID}-${standardDateFormat(
-                        new Date()
-                    )}.matched.txt`
-                ),
-                matchedSongs
-                    .map((x) => `${x.songName} - ${x.artistName}`)
-                    .join("\n")
-            );
+                    `${playlistID}-${standardDateFormat(new Date())}.txt`
+                );
+
+                await fs.promises.writeFile(
+                    playlistUnmatchedSongsPath,
+                    unmatchedSongs.join("\n")
+                );
+            }
+
+            if (KmqConfiguration.Instance.persistMatchedSpotifySongs()) {
+                await fs.promises.writeFile(
+                    path.resolve(
+                        SPOTIFY_PLAYLIST_UNMATCHED_SONGS_DIR,
+                        `${playlistID}-${standardDateFormat(
+                            new Date()
+                        )}.matched.txt`
+                    ),
+                    matchedSongs
+                        .map((x) => `${x.songName} - ${x.artistName}`)
+                        .join("\n")
+                );
+            }
         }
 
         matchedSongs = _.uniqBy(matchedSongs, "youtubeLink");
+        const metadata = {
+            playlistID,
+            playlistLength: spotifyMetadata.songCount,
+            playlistName: spotifyMetadata.playlistName,
+            matchedSongsLength: matchedSongs.length,
+            thumbnailUrl: spotifyMetadata.thumbnailUrl as string,
+        };
+
+        State.cachedSpotifyPlaylists[playlistID] = {
```

Comment:
nit: we're updating this cache even if the entry was originally retrieved from cache